include/xgboost/linalg.h

```diff
@@ -530,17 +530,17 @@ class TensorView {
   /**
    * \brief Number of items in the tensor.
    */
-  LINALG_HD [[nodiscard]] std::size_t Size() const { return size_; }
+  [[nodiscard]] LINALG_HD std::size_t Size() const { return size_; }
```

Comment:
clangd is not quite happy about the place of c++ attribute when running in CUDA mode.